File: neetcode-250/arrays-&-hashing/169-majority-element.ts
function majorityElement(nums: number[]): number {
	let numCount = new Map<number, number>();
	let res = 0,
		resCount = 0;
	for (let n of nums) {
		const newCount = (numCount.get(n) ?? 0) + 1;
		numCount.set(n, newCount);
		if (newCount > resCount) {
			res = n;
			resCount = newCount;
		}
	}
	return res;
}

File: neetcode-250/arrays-&-hashing/169-majority-element.py
class Solution:
    def majorityElement(self, nums: list[int]) -> int:
        maj, count, i = nums[0], 1, 1
        while i < len(nums):
            if nums[i] == maj:
                count += 1
            elif count == 1:
                maj = nums[i]
            else:
                count -= 1
            i += 1
        return maj
